src/modules/aws_ecr/rpcs/build.ts

```diff
@@ -164,31 +89,44 @@ export class EcrBuildRpc extends RpcBase {
       githubPersonalAccessToken ? '' : githubRepoUrl,
       githubRef,
     );
-    await this.createCodebuildProject(
+    const codeBuildProject = await this.createCodebuildProject(
       codeBuildProjectName,
       buildSpec,
       githubRef,
       githubPersonalAccessToken ? githubRepoUrl : '',
       role,
-      client,
+      region,
+      ctx,
     );
 
     // start build and wait for it to complete
-    const build = await this.startCodebuildProject(codeBuildProjectName, client);
-    await this.waitForBuildsToComplete(client.cbClient, [build.id!]);
+    const buildResult = await awsCodebuildModule.startBuild.call(
+      _dbId,
+      _dbUser,
+      ctx,
+      codeBuildProjectName,
+      region,
+    );
+    if (!buildResult.length || buildResult[0].status !== 'OK') {
+      throw new Error(`Error when starting codebuild project: ${buildResult[0].message}`);
+    }
 
     // get the pushed image from the cloud and create it in the DB
     const createdImage = await this.getCreatedEcrImage(ctx, ecrRepository);
-    this.module.repositoryImages.db.create(createdImage, ctx);
+    await this.module.repositoryImages.db.create(createdImage, ctx);
 
     // delete service role
-    awsIamModule.role.cloud.delete(role, ctx);
+    await awsIamModule.role.cloud.delete(role, ctx);
 
     // delete credentials
-    if (githubPersonalAccessToken) await this.deleteGithubCredentials(credentialsArn, client);
+    if (githubPersonalAccessToken)
+      await awsCodebuildModule.sourceCredentialsList.cloud.delete(
+        { arn: credentialsArn } as SourceCredentialsList, // hacky but works
+        ctx,
+      );
```

Comment:
can we add the braces here to avoid confusion?

Reply:
Can you clarify this a bit?

Reply:
If you mean parentheses, the linter will remove them once added.